lib/db.js

```diff
@@ -60,35 +63,76 @@ function normalizeType(params) {
   return type;
 }
 
-class LimitDB {
+class LimitDB extends EventEmitter {
   constructor(params) {
+    super();
     params = params || {};
 
     if (typeof params.path !== 'string' && !params.inMemory) {
       throw new Error('path is required');
     }
-    const path = !params.inMemory ? params.path : undefined;
 
-    this._db = level(path, {
+    const dbPath = !params.inMemory ? params.path : undefined;
+
+    const done = (err, db, types) => {
+      if (err) {
+        return this.emit('error', err);
+      }
+      this._db = db;
+      this._types = types;
+      this.emit('ready');
+    };
+
+    if (!dbPath) {
+      this._openDb(undefined, params, done);
+    } else {
+      this._safeOpenDb(dbPath, params, done);
+    }
+  }
+
+  _safeOpenDb(dbPath, params, done) {
+    dbChecker.check(dbPath, (err) => {
+      if (err) {
+        this.emit('repairing');
+        return leveldown.repair(dbPath, (err) => {
+          if (err) {
+            this.emit('error', err);
+            return;
+          }
+          this._safeOpenDb(dbPath, params, done);
```

Comment:
Won't this potentially create a loop and if we keep trying and failing also an stack overflow because of the lack of tail call optimization in node?

Reply:
yes it will.. I will call directly to _openDb